docs/infrastructure/microk8s-quickstart.md

```diff
@@ -0,0 +1,325 @@
+
+## 1. Concepts
+
+### Kubeconfig
+
+Any tool that operates on top of a cluster needs to know which cluster it is going to work on. This execution context is typically called the `kubeconfig` which is stored in the `~/.kube/config` file. This file basically defines a list of clusters and it holds several pieces of information per cluster entry:
+
+* The address of the cluster (e.g. https://104.248.34.242:164309).
+* The user friendly name of the cluster (e.g. microk8s-cluster).
+* A reference to the current context. This is the active cluster against which any kubernetes control (`kubectl`) command you run will get executed.
+
+### Helm and tiller
+
+Two executables will get installed onto the cluster as part of this process: `helm` and `tiller`.
+
+* `helm` is responsible for resolving the configuration of the chart to be installed, while issuing the correct install commands onto the cluster.
+* `tiller` is responsible for receiving the install commands issued by `helm`, as well as managing the lifecycle of the components that have been installed.
+
+`helm` is the client facing side, while `tiller` is the server/cluster facing side.
+
+## 2. Prepare your environment
+
+Prepare your environment to set up the microk8s cluster:
+
+1. A machine running Ubuntu 18.04 LTS.
+2. Establish a remote SSH session onto this machine.
+
+Assuming that you are starting from a blank slate, the first step is to install microk8s. Otherwise, please jump on to the section [Configuring microk8s](###4.-Configuring-microk8s).
+
+## 3. Installing microk8s
+
+1. Make sure the machine is up to date and also has the `nfs-common` package installed
+
+   ```bash
+   sudo apt update && apt install nfs-common -y
+
+2. Install microk8s
+   >sudo snap install microk8s --classic --channel=1.15/stable
+   1. We recommend following the official documentation on how to [install microk8s](https://microk8s.io/docs/).
+
+## 4. Configuring microk8s
+
+1. First, we must enable the following plugins on microk8s:
+    >sudo microk8s.enable dns && sudo microk8s.status --wait-ready
+
+    >sudo microk8s.enable storage && sudo microk8s.status --wait-ready
+
+    >sudo microk8s.enable ingress &&  sudo microk8s.status --wait-ready
+
+2. Now that these plugins have been enabled, we should restart microk8s to ensure a proper bootstrap of the cluster
+    >microk8s.stop && microk8s.start && microk8s.status --wait-ready
+
+    Further information on this subject can be [found here](https://github.com/ubuntu/microk8s/issues/493#issuecomment-498167435).
+3. The plugins are now enabled and the cluster bootstrapped. However, we must still wait for some microk8s internals (dns, http, and ingress) plugins to be ready. Failing to do so can result in pods entering a `CrashLoopBackoff` state:
+
+    >microk8s.kubectl wait -n kube-system --for=condition=Ready pod -l k8s-app=kube-dns --timeout=300s
```

Comment:
I don't think so because one is waiting for kube-system, and the other is waiting microk8s itself :/
as for the timeouts, you are absolutely right. removing them :)